backend/api/handlers/semester/PostSemesterHandler.go

```diff
@@ -0,0 +1,56 @@
+package semester
+
+import (
+	"fmt"
+	"net/http"
+
+	"LiteracyLink.com/backend/api/model"
+	"LiteracyLink.com/backend/db"
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+func PostSemesterHandler(c *gin.Context) {
+
+	var semest model.Semester
+
+	// Bind the request body to the Semester struct
+	if err := c.ShouldBindJSON(&semest); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	// Validate spring or fall value
+	if !isValidSemester(semest.SpringOrFall) {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid value for Spring or fall"})
+		return
+	}
+
+	// Now you use the semest to create a new Semester in the database
+	dbc := c.MustGet("db").(*gorm.DB)
+	err := db.CreateSemester(semest, dbc)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"status":  "failed",
+			"message": fmt.Sprintf("ERROR: %g", err),
+		})
+		return
+	}
+
+
+	c.JSON(http.StatusCreated, gin.H{})
+}
+
+
+func isValidSemester(sem string) bool {
+
+	//TODO: accept all caps too? 
```

Comment:
Please complete

Reply:
I don't need to because you are going to do a drop down that only send me "spring" or "fall" ;)